helpers/checkpoint_logger/__init__.py

```diff
@@ -432,15 +436,22 @@ def submit_subflow(self: _Self, metric: str, start: T, end: T) -> _Self:
         if duration:
             sentry_sdk.set_measurement(metric, duration, "milliseconds")
             duration_in_seconds = duration / 1000
+            repoid = self.context and self.context.repoid
             PROMETHEUS_HANDLER.log_subflow(
-                flow=self.cls.__name__, subflow=metric, duration=duration_in_seconds
+                flow=self.cls.__name__,
+                subflow=metric,
+                duration=duration_in_seconds,
+                repoid=repoid,
             )
 
         return self
 
 
 def from_kwargs(
-    cls: type[T], kwargs: MutableMapping[str, Any], strict: bool = False
+    cls: type[T],
+    kwargs: MutableMapping[str, Any],
+    strict: bool = False,
+    context: CheckpointContext | None = None,
```

Comment:
So I had to play with this a bit to make it work. The `kwargs.get(_kwargs_key(cls), {}).copy()` is an empty object the first time a flow is ran, so instead I did 

``` python

def from_kwargs(
    cls: type[T],
    kwargs: MutableMapping[str, Any],
    strict: bool = False,
) -> CheckpointLogger[T]:
    context = kwargs.pop("context", CheckpointContext())
    data = kwargs.get(_kwargs_key(cls), {})

    # kwargs has been deserialized into a Python dictionary, but our enum values
    # are deserialized as simple strings. We need to ensure the strings are all
    # proper enum values as best we can, and then downcast to enum instances.
    deserialized_data = {}
    deserialized_data["context"] = context

    for checkpoint, timestamp in data.items():

```

it could be prettier but that's what I could think, wdyt?